dashboard/app/controllers/projects_controller.rb

```diff
@@ -380,6 +379,15 @@ def set_level
 
   private
 
+  # Helper function to keep the animation and source remix actions together and ordered
+  # Source remix assumes the animations are already copied
+  # @param src_channel_id [String]
+  # @param new_channel_id [String]
+  def remix_animation_and_sources(src_channel_id, new_channel_id)
+    AnimationBucket.new.copy_files src_channel_id, new_channel_id
+    SourceBucket.new.remix_source src_channel_id, new_channel_id
+  end
```

Comment:
I'm not sure how helpful this private helper is.

The soft enforcement provided by the method extraction here isn't much stronger than a comment inside the `remix` method would be.  Stronger enforcement is probably provided by tests anyway, and if you want very strong enforcement we should set things up so that the `AnimationBucket` operation returns some value that the `SourceBucket` operation requires, like a map of updated version ids.

That said, I'm also rethinking where the remix work should live.  We [already gave SourceBucket knowledge of AnimationBucket](https://github.com/code-dot-org/code-dot-org/blob/staging/shared/middleware/helpers/source_bucket.rb#L38-L52) which isn't elegant but gets the job done, so there's no reason we couldn't move the `AnimationBucket` call here into `SourceBucket::remix_source`.  On the other hand, it's probably cleaner for `ProjectsController` to own all multi-bucket operations, so maybe the right thing to do is to hoist some of the logic in `remix_source` up into `ProjectsController::remix`.